### ISSUE TITLE
Sync Cache Sometimes Ignored

### ISSUE THREAD

**Issue by apenngrace** [s3tools/s3cmd#452]
(I am using 1.5.0-rc1)  When running the `s3cmd sync` command, if I run with the `--cache-file=FILE` option and I do not use `--verbose` or `--debug` options, then s3cmd will rebuild the cache file each time (about 4 hours for 120GB).  

However, if I do set either `--verbose` or `--debug` modes then it seems to use the cache and will begin to the S3 file listing within approximately 10 minutes (instead of 4 hours).

Seems like a bug.

**Comment by mdomsch** [comment]
I don't believe this is true.

$ ./s3cmd --cache-file=/tmp/cache.pkl sync . s3://s3cmd-test.domsch.com/
builds the cache file including reading the files, calculating the md5sums.

On subsequent runs, watching ltrace, it only stat()s all the files.

--verbose and --debug only change the output loglevel; there are no code path changes as a result of those flags.

**Comment by apenngrace** [comment]
Maybe there is something unexpected happening.  

I spent a couple of days playing with this, and those flags seemed to influence the behavior I was seeing.  I saw the behavior I am describing pretty consistently (over several days of testing).  I am using a raspberry pi using a cifs mount to connect to a Drobo share.  The data is being copied from the raspberry pi to s3 using s3cmd.

This issue would not be too much of a problem if the process would complete successfully.  I keep getting a broken pipe error on the exact same file (tried it different days, different weeks, same thing).  So unfortunately, I have never gotten the 120GB sync to work for me.  I saved a log in a text file if you think it might be useful to look at.  Does the broken pipe happen when Amazon stops the connection?
